### ISSUE TITLE
1.14.1 "Export into sysex files" dialog is not working

### ISSUE THREAD

**Issue by Andy2No** [christofmuc/KnobKraft-orm#143]
When I click the "Export into sysex files" button, I get the dialog box to choose options, but when I then click Export, nothing happens.

This is the case for the library tab for each synth I've tried so far - without the synth connected, but the source is the database, so it seems like this should work... and I seem to remember it used to.

Nothing is shown in the log window.

I think this problem did already exist in 1.14.0.  I was trying to do it to test an adaptation I was making, and it wasn't working.  I thought maybe it was due to a bug I'd introduced which was corrupting patches when converting them between types - that was what I was trying to use it to figure out, by exporting patches to look at them.  However, I see it now does it for the patches for all synths I've tried.

**Comment by Andy2No** [comment]
I've now confirmed that this already was the case in 1.14.0.  That also applies to #142 

I tried reverting to 1.13.6 to test it in that, but it complains that it can't open the database, which I'd saved in a folder in My Documents.... Presumably I'd need to edit something in the registry to stop it trying to open that database, so I could try it.

This is on Win 7 SP1 x64.

**Comment by christofmuc** [comment]
@Andy2No The databse that is opened is stored in the %APPDATA%\KnobKraftOrm\KnobKraftOrm.settings file, which is XML and can be edited. I believe if you delete the "LastDatabase" key it will start up without trying to open the incompatible new one, and you can browse to e.g. the backup version with the old database format, which is archived with a "-before-migration" postfix in its name. 

Seems the list tree might have broken something in the import/export, I'll look into it.

**Comment by Andy2No** [comment]
Thanks.  FWIW, I've also had to rename C:\Users\andy\AppData\Roaming\knobkraft\SysexDatabaseOfAllPatches.db3 to get it to not try to open that one, and make a new one.

The "Import sysex files from computer" button isn't working in 1.13.6 either.  I was able to load a backup database, and found that "Export into sysex files" doesn't work either.

The same applies to 1.13.2 - neither button works... Ditto 1.13.1, 1.13.0 and 1.12.1

This is odd, because I've definitely used both features before.  I successfully exported some patches from the Tetra, using the custom adaptation I was working on, on the 25th of January, going by the file dates.  That seems to imply I was using 1.13.6 at the time.

Since none of the much earlier versions of Orm work either, I'm now wondering if it's been caused by updating the VC runtime, or perhaps having had a newer version of Python, briefly?  I'm not clear where that's installed to.  Perhaps I need to do a more thorough uninstallation, to successfully revert to an earlier version?

I see at least part of python in C:\Users\andy\AppData\Local\Programs\KnobKraftOrm, after isntalling 1.12.1.

I'll try deleting everything I can find, and installing 1.13.6 again.

**Comment by Andy2No** [comment]
Okay, I've got both buttons working again in 1.13.6, still without a synth connected, by doing a cleaner uninstall.

After uninstalling 1.12.1, I deleted, IIRC;

C:\Users\andy\AppData\Roaming\KnobKraft
C:\Users\andy\AppData\Roaming\KnobKraftOrm

and

C:\Users\andy\AppData\Local\Programs\KnobKraftOrm

I didn't empty my Adaptations folder, but I did back it up, just in case.

I then installed 1.13.6 again and both "Export into sysex files" and "Import sysex files from computer" are now working again.

So, yes, it looks like what's needed is a cleaner uninstall process, to be able to successfully revert to an earlier version.

I think I'll leave it like that, for now, remaining on 1.13.6, so I can have another go at the Tetra adaptation, but let me know if there's anything else you'd like me to try.
